### ISSUE TITLE
Exposing CommonFunctions as public methods on SWCompression

### ISSUE THREAD

**Issue by agirault** [tsolomko/SWCompression#19]
The main reason why I'm looking into this library is to run `untar` on iOS.

This is shown here:
https://github.com/tsolomko/SWCompression/blob/b5c629654f70e7f219e635e97a4d4657633b5ee1/Sources/swcomp/Containers/TarCommand.swift#L61-L62

However, `swcomp` is not exposed when importing `SWCompression`, so the [`write`](https://github.com/tsolomko/SWCompression/blob/b5c629654f70e7f219e635e97a4d4657633b5ee1/Sources/swcomp/Containers/CommonFunctions.swift#L28) method which I need is inaccessible.

Could swcomp be exposed, or could the `write` method be added to the `Container` class (signature example below)?
```swift
public protocol Container {

    ...

    /// Write the entries on the filesystem
    static func write(entries: [Entry], to url: URL) throws
}
```

**Comment by tsolomko** [comment]
Hmm, I am not sure about this.

It was an explicit design decision to not provide APIs for actually working with filesystem. Partially, this is because there are many difficulties to be aware of when implementing such functionality. For example, there can be differences in how things should work on different platforms with different file systems. In addition, when you actually write files to file system you need to implement measures against malicious archives such as [tarbomb](https://en.wikipedia.org/wiki/Tar_(computing)#Tarbomb).

`swcomp` was implemented as a proof-of-concept and to provide an example of how SWCompression can be used in practice, and as such it is likely that its implementation lacks features outlined above.

To sum up, there are no current plans neither to expose `swcomp` nor add the corresponding functionality to SWCompression directly.

**Comment by agirault** [comment]
Understood, thank you.
